python/knext/knext/examples/musique/run_experiment.py

```diff
@@ -0,0 +1,279 @@
+
+
+CWD = os.path.dirname(os.path.abspath(__file__))
+WORKSPACE = os.path.join(CWD, 'workspace')
+if not os.path.exists(WORKSPACE):
+    os.makedirs(WORKSPACE)
+DATA_DIR = '/Users/lisiyuan/Codes/FromLLMtoAgent/data/musique-main/data'
+
+
+def get_official_deepseek_llm():
+    client_config = {
+        'model_name' : 'deepseek-chat',
+        'base_url' : 'https://api.deepseek.com',
+        'api_key' : 'API-KEY',
+    }
+
+    llm = get_llm_client(
+        client_name = 'openai',
+        client_config = client_config,
+    )
+    return llm
+
+
+def get_official_aliyun_embed_fn():
+    return None
+
+
+
+
+def create_musique_spo_info_tools(musique_data, get_llm_fn, get_embed_fn, prompt_template_dir, intermediate_dir):
+    spo_retriver = SPOInfoRetriverWithTripleExtractor(
+        musique_dataset=musique_data, 
+        llm=get_llm_fn(), 
+        embedding_fn=get_embed_fn(), 
+        prompt_template_dir=prompt_template_dir,
+        intermediate_dir=intermediate_dir,
+    )
+    return spo_retriver
+
+
+def create_musique_hierarchical_info_tools(musique_data, get_llm_fn, get_embed_fn, prompt_template_dir, intermediate_dir):
+    info_tools = []
+    spo_retriver = SPOInfoRetriverWithTripleExtractor(
+        musique_dataset=musique_data, 
+        llm=get_llm_fn(), 
+        embedding_fn=get_embed_fn(), 
+        prompt_template_dir=prompt_template_dir,
+        intermediate_dir=intermediate_dir,
+    )
+    info_tools.append(spo_retriver)
+    
+    text_retriver = TextInfoRetriver(
+        musique_dataset=musique_data, 
+        llm=get_llm_fn(), 
+        embedding_fn=get_embed_fn(), 
+        top_k=3
+    )
+    info_tools.append(text_retriver)
+    return info_tools
+
+
+def create_divide_and_conquer_agent_v1(musique_data, get_llm_fn, get_embed_fn, prompt_template_dir, intermediate_dir, debug_mode):
+    # spo answer
+    llm = get_llm_fn()
+    
+    spo_info_tool = create_musique_spo_info_tools(
+        musique_data=musique_data, 
+        get_llm_fn=get_llm_fn, 
+        get_embed_fn=get_embed_fn, 
+        prompt_template_dir=prompt_template_dir, 
+        intermediate_dir=intermediate_dir
+    )
+        
+    answer_parent_question = AnswerQuestionWithContext(
+        llm_module=llm, 
+        use_default_prompt_template=False,
+        prompt_template_dir=prompt_template_dir
+    )
+    
+    atom_answer_question = MusiqueAnswerQuestionWithSPO(
+        llm_module=llm,
+        spo_info_tool=spo_info_tool,
+        prompt_template_dir=prompt_template_dir,   
+    )
+    
+    agent = MusiqueDivideAndConquerAgent(
+        llm=llm,
+        answer_parent_question=answer_parent_question,
+        answer_atom_question=atom_answer_question,
+        prompt_template_dir=prompt_template_dir,
+        debug_mode=debug_mode
+    )
+    return agent
+
+
+def create_divide_and_conquer_agent_v2(musique_data, get_llm_fn, get_embed_fn, prompt_template_dir, intermediate_dir, debug_mode):
```

Comment:
what is the difference between `create_divide_and_conquer_agent_v1` and `create_divide_and_conquer_agent_v2`?